rholang/reference_doc/bundles/README.md

```diff
@@ -1,22 +1,48 @@
 ## Bundles
-Bundles can be used to control the read/write privileges on a channel
-
-### Write only
-- `bundle+{ chan }`
-- `chan` is write only
-
-### Read only
-- `bundle-{ chan }`
-- `chan` is read only
-
-### No read, no write
-- `bundle0{ chan }`
-- `chan` can't be read or written on
-- Useful in some cases
-  - EG here
-
-### bundle: bundle read write
-- `bundle{ chan }`
-- `chan` can be read and written on
-- This is useful for other circumstances
-  - EG here
+Bundles can be used to control the read/write privileges on a channel. They are most applicable and useful when several processes are communicating through the same channel. For example, you'd like for other people to write you emails, not read your emails! Also, bundles cannot be deconstructed through pattern matching.
+
+### Write-only
+The syntax for a *write-only* bundle is: `bundle+{P}`.
+
+Processes with only a handle on `bundle+{P}` (and not `P`) can send messages to the channel `@P`, but they cannot consume data from `@P`. To send a message to `@P`, one simply does `@{bundle+{P}}!(Msg)`. However, `for(x <- @{bundle+{P}}){Cont}` returns the error: `Trying to read from non-readable channel`. The `+` means you can only add to the channel, not subtract!
+
+A process with a handle on the underlying process `P` can, of course, read messages sent to the channel `@P`.
+
+```rholang
+  new a, out(`rho:io:stdout`) in {
+    @{bundle+{*a}}!(Nil) |
+    for (_ <- a) { out!("received") }
+  }
+
+// prints to stdout: "received"
+```
+
+### Read-only
+The syntax for a *read-only* bundle is: `bundle-{P}`.
+
+Processes with only a handle on `bundle-{P}` (and not `P`) can receive messages on `@P`, but they cannot send messages on `@P`. To consume a message from `@P`, one simply does `for(x <- @{bundle-{P}}){Cont}`. However, `@{bundle-{P}}!(Msg)` returns the error: `Trying to send on non-writeable channel`. The `-` means you can only subtract from the channel, not add!
+
+A process with a handle on the underlying process `P` can, of course, write messages to the channel `@P`.
+
+```rholang
+  new a, out(`rho:io:stdout`) in {
+    a!(Nil) |
+    for (_ <- @{bundle-{*a}}) { out!("sent") }
+  }
+
+// prints to stdout: "sent"
+```
+
+### No-Read/No-Write
+The syntax for a *no-read/no-write* bundle is: `bundle0{P}`.
+
+Processes with only a handle on `bundle0{P}` (and not `P`) cannot read from or write to the channel `@P`. The `0` means nothing can be added or taken.
```

Comment:
I'm curious what's the use case for this?

Reply:
It's used in several Casper contracts:
https://github.com/rchain/rchain/blob/d61efd6693462914a163d36b295598184b6fb737/casper/src/main/resources/RevVault.rho#L165
https://github.com/rchain/rchain/blob/d61efd6693462914a163d36b295598184b6fb737/casper/src/main/resources/AuthKey.rho#L53
https://github.com/rchain/rchain/blob/d61efd6693462914a163d36b295598184b6fb737/casper/src/main/resources/MakeMint.rho#L36
It looks like it's used for things the system needs to keep track of and should not be deconstructed by users.